File: VB/WebApplication1/Controllers/HomeController.vb
Imports DevExpress.Web.Mvc
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc
Imports WebApplication1.Code
Imports WebApplication1.Models

Namespace WebApplication1.Controllers
    Public Class HomeController
        Inherits Controller

        Public Function Index() As ActionResult
            Return View()
        End Function

        Public Function TreeListPartial() As ActionResult
            Return PartialView("_TreeListPartial", LicensesDataProvider.GetLicenses())
        End Function
        Public Function TreeListCustomPartial(ByVal textsString As String) As ActionResult
            ViewBag.Selection = textsString
            Return PartialView("_TreeListPartial", LicensesDataProvider.GetLicenses())
        End Function

        Public Function GridViewPartial() As ActionResult
            Return PartialView("_GridViewPartial", CustomersDataProvider.GetCustomers())
        End Function
        <HttpPost, ValidateInput(False)>
        Public Function GridViewAddNewPartial(ByVal customer As Customer) As ActionResult
            CustomersDataProvider.InsertCustomer(customer)
            Return GridViewPartial()
        End Function
        <HttpPost, ValidateInput(False)>
        Public Function GridViewUpdatePartial(ByVal customer As Customer) As ActionResult
            Dim dropDownEditText = DropDownEditExtension.GetValue(Of String)("DropDownEdit1")
            Dim fieldValue = ValueAndDisplayTextHelper.GetValueString(dropDownEditText)
            customer.Licenses = fieldValue
            CustomersDataProvider.UpdateCustomer(customer)
            Return GridViewPartial()
        End Function
        <HttpPost, ValidateInput(False)>
        Public Function GridViewDeletePartial(Optional ByVal ID As Integer = -1) As ActionResult
            If ID <> -1 Then
                CustomersDataProvider.DeleteCustomer(New Customer With {.CustomerID = ID})
            End If
            Return GridViewPartial()
        End Function
    End Class
End Namespace

File: CS/WebApplication1/Controllers/HomeController.cs
using DevExpress.Web.Mvc;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using WebApplication1.Code;
using WebApplication1.Models;

namespace WebApplication1.Controllers {
    public class HomeController : Controller {
        public ActionResult Index() {
            return View();
        }

        public ActionResult TreeListPartial() {
            return PartialView("_TreeListPartial", LicensesDataProvider.GetLicenses());
        }
        public ActionResult TreeListCustomPartial(string textsString) {
            ViewBag.Selection = textsString;
            return PartialView("_TreeListPartial", LicensesDataProvider.GetLicenses());
        }
        public ActionResult GridViewPartial() {
            return PartialView("_GridViewPartial", CustomersDataProvider.GetCustomers());
        }
        [HttpPost, ValidateInput(false)]
        public ActionResult GridViewAddNewPartial(Customer customer) {
            CustomersDataProvider.InsertCustomer(customer);
            return GridViewPartial();
        }
        [HttpPost, ValidateInput(false)]
        public ActionResult GridViewUpdatePartial(Customer customer) {
            var dropDownEditText = DropDownEditExtension.GetValue<string>("DropDownEdit1");
            var fieldValue = ValueAndDisplayTextHelper.GetValueString(dropDownEditText);
            customer.Licenses = fieldValue;
            CustomersDataProvider.UpdateCustomer(customer);
            return GridViewPartial();
        }
        [HttpPost, ValidateInput(false)]
        public ActionResult GridViewDeletePartial(int ID = -1) {
            if (ID != -1)
                CustomersDataProvider.DeleteCustomer(new Customer { CustomerID = ID });
            return GridViewPartial();
        }
    }
}